subprojects/soak/src/integTest/groovy/org/gradle/jvm/toolchain/JavaToolchainDownloadSoakTest.groovy

```diff
@@ -79,6 +82,41 @@ class JavaToolchainDownloadSoakTest extends AbstractIntegrationSpec {
         assertJdkWasDownloaded("openj9")
     }
 
+    def "clean destination folder when downloading toolchain"() {
+        when:
+        result = executer
+                .withTasks("compileJava", "-Porg.gradle.java.installations.auto-detect=false")
+                .expectDocumentedDeprecationWarning("Java toolchain auto-provisioning needed, but no java toolchain repositories declared by the build. Will rely on the built-in repository. " +
+                        "This behavior has been deprecated. This behavior is scheduled to be removed in Gradle 8.0. " +
+                        "In order to declare a repository for java toolchains, you must edit your settings script and add one via the toolchainManagement block. " +
+                        "See https://docs.gradle.org/current/userguide/toolchains.html#sec:provisioning for more details.")
+                .run()
+
+        then:
+        javaClassFile("Foo.class").assertExists()
+        assertJdkWasDownloaded("adoptopenjdk")
+
+        when:
+        //delete marker file to make the previously downloaded installation undetectable
+        def markerFile = findMarkerFile(executer.gradleUserHomeDir.file("jdks"))
+        markerFile.delete()
+
+        then:
+        !markerFile.exists()
```

Comment:
Suggestion: It seems odd that this would be in a `then` block as it's not really testing any functionality, just asserting a precondition.  You might consider making this just `assert markerFile.delete()` and then collapse the next `when` into an `and` block.  I think that would make for a cleaner stimulus/response.

Reply:
Good point, have rewritten the test.